languages/javalang/builder-renderer/commercetools-java-client/src/main/java/client/middlewares/CtpOAuthMiddlewareForClientCredentialsGrant.java

```diff
@@ -0,0 +1,79 @@
+package client.middlewares;
+
+import client.Middleware;
+import client.MiddlewareArg;
+import com.github.scribejava.core.builder.ServiceBuilder;
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.oauth.OAuth20Service;
+
+import java.util.concurrent.CompletableFuture;
+
+public final class CtpOAuthMiddlewareForClientCredentialsGrant implements Middleware {
+
+    private final String clientId;
+    private final String clientSecret;
+    private final String accessTokenEndpoint;
+    private final String authorizationBaseUrl;
+    private final String scope;
+    private final OAuth20Service oAuth20Service;
+
+
+    public CtpOAuthMiddlewareForClientCredentialsGrant(String clientId, String clientSecret, String scope, String accessTokenEndpoint, String authorizationBaseUrl) {
+        this.clientId = clientId;
+        this.clientSecret = clientSecret;
+        this.accessTokenEndpoint = accessTokenEndpoint;
+        this.authorizationBaseUrl = authorizationBaseUrl;
+        this.scope = scope;
+        oAuth20Service = new ServiceBuilder(clientId)
```

Comment:
Hmm, in our old SDK we don't rely on an external oauth2 client. And we always try to use as few as possible external dependcies... Do you know how many transitive dependencies this pulls in?

Reply:
Not sure about the number but the idea here is that the middleware can pull in as much as needed since worst case scenario we create a new middlware with new dependencies.